test/types/types.test.ts

```diff
@@ -61,21 +62,24 @@ const options2 = {
   global: true,
   max: (req: FastifyRequest<RequestGenericInterface>, key: string) => 42,
   allowList: (req: FastifyRequest<RequestGenericInterface>, key: string) => false,
-  timeWindow: 5000
+  timeWindow: 5000,
+  hook: 'preParsing' as RateLimitHook
```

Comment:
Why would `expectAssignable` be useful here?
Since I'm enumerating valid hook options, directly using literal string fires:

```console
Type 'string' is not assignable to type 'RateLimitHook | undefined'.
```

To avoid the workaround with ` as RateLimitHook` (😬) I can explicitly set `let hook: RateLimitHook` before options objects.

Reply:
Oups my bad I forgot to add `const option2: RateLimitPluginOptions = ...` and same goes for option3 and option4.

With this it will even trigger a typescript error if the value of hook is not valid.

Reply:
ooh yes! truth 🙃